### PR TITLE
fix: Check regions

### LINKED ISSUE #583: Provider Hangs if Region name is misspelled
### Describe the bug

Provider hangs if region specified in `config.hcl` is not actually a valid region

### Expected Behavior

Should throw an error

### Steps to Reproduce

run fetch with this config:
``` hcl
provider "aws" {
  configuration {
    regions = ["us-eadst-2"]
  }
  resources = ["*"]
}

```

### Possible Solution

_No response_

### Provider and CloudQuery version

all?

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by zagronitay**

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -268,6 +273,9 @@ func verifyRegions(regions []string) error {
 		if i > 0 && hasWildcard {
 			return errInvalidRegion
 		}
+		if ok := availableRegions[region]; !ok {
+			return fmt.Errorf("unknow region: %s", region)
```
Comment (disq) [inline client/client.go:277]:
```suggestion
			return fmt.Errorf("unknown region: %q", region)
```

File: client/helpers.go [hunk 2]
```diff
@@ -91,6 +91,32 @@ func isSupportedServiceForRegion(service string, region string) bool {
...
 
+func getAvailableRegions() (map[string]bool, error) {
+	readOnce.Do(func() {
+		supportedServiceRegion = readSupportedServiceRegions()
+	})
+
+	regionsSet := make(map[string]bool)
+
+	if supportedServiceRegion == nil {
+		return nil, fmt.Errorf("cloud not get AWS regions/services data")
```
Comment (disq) [inline client/helpers.go:102]:
```suggestion
		return nil, fmt.Errorf("could not get AWS regions/services data")
```

File: client/helpers.go [hunk 3]
```diff
@@ -91,6 +91,32 @@ func isSupportedServiceForRegion(service string, region string) bool {
...
+	}
+
+	if supportedServiceRegion.Partitions == nil {
+		return nil, fmt.Errorf("cloud not found any AWS partitions")
```
Comment (disq) [inline client/helpers.go:106]:
```suggestion
		return nil, fmt.Errorf("could not find any AWS partition")
```

**Review by disq — CHANGES_REQUESTED** [review]

**Comment by bbernays** [comment]
Could you add a test for this change?

**Review by roneli — APPROVED** [review]
LGTM, would add a tests as suggest by @bbernays

**Review by disq — APPROVED** [review]

**Review by bbernays — APPROVED** [review]
LGTM

**Pull request merged**